### PR TITLE
DEV: dropped unused post_details

### PR DESCRIPTION
**Pull request by jjaffeux**
`post_details` have been replaced by `post_custom_fields` and never got replaced.

### CODE REVIEW

File: db/migrate/20200601130900_correct_schema_discrepancies.rb [hunk 1]
```diff
@@ -98,8 +98,6 @@ def up
       plugin_store_rows.key
       plugin_store_rows.plugin_name
       plugin_store_rows.type_name
-      post_details.key
-      post_details.value
```
Comment (jjaffeux) [inline db/migrate/20200601130900_correct_schema_discrepancies.rb:102]: I have doubts about this

**Comment by davidtaylorhq** [comment]
How sure are you that no plugins are using this? I wonder if we should add a deprecation notice to the model, and then wait a while before dropping the table. (like we are doing for https://github.com/discourse/discourse/blob/main/app/models/oauth2_user_info.rb)

**Comment by jjaffeux** [comment]
> How sure are you that no plugins are using this? I wonder if we should add a deprecation notice to the model, and then wait a while before dropping the table. (like we are doing for https://github.com/discourse/discourse/blob/main/app/models/oauth2_user_info.rb)

Im pretty sure it has never been used. `post_custom_fields` has been added 8 years ago, and the original commit of `post_details` don't even use it itself. Most likely it's been pre-mature engineering.

https://github.com/discourse/discourse/commit/2450088c03719db0c17e71dfc44f48a7862b74b3

---

This being said, Im not against the deprecate, it's certainly safer.

**Comment by jjaffeux** [comment]
@SamSaffron By any chance do you have any memory of this?

**Comment by jjaffeux** [comment]
Will close this for now

**Pull request closed without merging**